src/pairings.rs

```diff
@@ -533,12 +537,20 @@ impl From<G2Affine> for G2Prepared {
     }
 }
 
-#[cfg(feature = "alloc")]
-#[cfg_attr(docsrs, doc(cfg(all(feature = "pairings", feature = "alloc"))))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
```

Comment:
Just to ensure we don't accidentally compare these where they shouldn't be:
```suggestion
#[derive(Clone, Copy, Debug)]
```